isso/ext/notifications.py

```diff
@@ -139,17 +139,42 @@ def format(self, thread, comment, parent_comment, recipient=None, admin=False):
 
         else:
             uri = self.public_endpoint + "/id/%i" % parent_comment["id"]
-            key = self.isso.sign(('unsubscribe', recipient))
+            key = self.isso.sign(('unsubscribe', recipient_comment["email"]))
 
-            rv.write("Unsubscribe from this conversation: %s\n" % (uri + "/unsubscribe/" + quote(recipient) + "/" + key))
+            rv.write("Unsubscribe from this conversation: %s\n" % (uri + "/unsubscribe/" + quote(recipient_comment["email"]) + "/" + key))
 
         rv.seek(0)
         return rv.read()
 
+    def notify_subject(self, thread, comment, parent_comment=None, recipient_comment=None, admin=False):
+        if admin:
+            return self.isso.conf.get("mail", "subject_admin").format(
+                title=thread["title"],
+                replier=comment["author"] or "Anonymous")
+        subject_format = list(self.isso.conf.getiter("mail", "subject_user"))
```

Comment:
What if subject_user is not set? What if there are more than 2 items in the list?

Reply:
Re the first question: Similar to the answer to `What happens if subject_admin is not set?` above.  
Re the second question: Only the first two items will be used.